File: src/leetcode/s01_100/leetcode_70.java
package leetcode.s01_100;

public class leetcode_70 {


    public int climbStairs(int n) {
        int[] arr = new int[n+1];
        arr[0] = 0;
        arr[1] = 1;
        arr[2] = 2;
        for(int i=3;i<n+1;i++) {
            arr[i] = arr[i-1]+arr[i-2];
        }
        return arr[n];
    }

    public static void main(String[] args) {
        leetcode_70 t = new leetcode_70();
        System.out.println(t.climbStairs(3));
    }
}

File: leetcode/s0001_0100/leetcode_70.py
from typing import *


class leetcode_70(object):
    """leetcode_70 documentation"""

    def __init__(self, *args, **kwargs):
        """Constructor for leetcode_70"""
        pass

    def solve(self):
        print(self.climbStairs(4))

    def climbStairs(self, n: int) -> int:
        a = [0, 1, 2]
        if n == 0 or n == 1 or n == 2:
            return a[n]
        for i in range(3, n+1):
            # print(i)
            a.append(a[i-1]+a[i-2])
        return a[n]



if __name__ == '__main__':
    l: leetcode_70 = leetcode_70()
    l.solve()
